From bce59c37991d04edac087ba78bf9a81c6bc4a63b Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Fri, 27 Feb 2015 08:33:10 +0100
Subject: [PATCH] iio: tsl2x7x: Fix trigger

Interrupt trigger is low and not rising.
Set 0x11 as default for Persistence Register 0xC.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 drivers/staging/iio/light/tsl2x7x_core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/iio/light/tsl2x7x_core.c b/drivers/staging/iio/light/tsl2x7x_core.c
index d64da7a..b2a9129 100644
--- a/drivers/staging/iio/light/tsl2x7x_core.c
+++ b/drivers/staging/iio/light/tsl2x7x_core.c
@@ -247,7 +247,7 @@ static const struct tsl2x7x_settings tsl2x7x_default_settings = {
 	.als_cal_target = 150,
 	.als_thresh_low = 200,
 	.als_thresh_high = 256,
-	.persistence = 255,
+	.persistence = 0x11,
 	.interrupts_en = 0,
 	.prox_thres_low  = 0,
 	.prox_thres_high = 512,
@@ -1906,7 +1906,7 @@ static int tsl2x7x_probe(struct i2c_client *clientp,
 		ret = request_threaded_irq(clientp->irq,
 					   NULL,
 					   &tsl2x7x_event_handler,
-					   IRQF_TRIGGER_RISING | IRQF_ONESHOT,
+					   IRQF_TRIGGER_LOW | IRQF_ONESHOT,
 					   "TSL2X7X_event",
 					   indio_dev);
 		if (ret) {
-- 
2.1.4

